cli/src/main/java/com/devonfw/tools/ide/git/repository/RepositoryProperties.java

```diff
@@ -0,0 +1,110 @@
+package com.devonfw.tools.ide.git.repository;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.devonfw.tools.ide.cli.CliException;
+import com.devonfw.tools.ide.context.IdeContext;
+
+/**
+ * {@link Properties} for {@link RepositoryConfig}.
+ */
+final class RepositoryProperties {
+
+  private final Path file;
+
+  private final Properties properties;
+
+  private final IdeContext context;
+
+  /**
+   * The constructor.
+   *
+   * @param file the {@link Path} to the properties file.
+   * @param context the {@link IdeContext}.
+   */
+  public RepositoryProperties(Path file, IdeContext context) {
+    this(file, context, context.getFileAccess().readProperties(file));
+  }
+
+  RepositoryProperties(Path file, IdeContext context, Properties properties) {
```

Comment:
Please add Javadoc, not clear why it is being done like this.

Reply:
Arent we going to far with this JavaDoc everywhere?
This constructor is neither `public` nor `protected` - but if it makes you happy, I will formally add this JavaDoc even though we have the same constructor above with its parameters documented already.

Reply:
Done.